### PR TITLE
kpt pkg cat command

### PR DESCRIPTION
**Pull request by phanimarupaka**
@frankfarzan 

This PR is to add `cat` command to `kpt pkg` group.

### CODE REVIEW

File: internal/docs/generated/pkgdocs/docs.go [hunk 1]
```diff
@@ -45,6 +45,18 @@ var PkgExamples = `
   $ kpt pkg update helloworld@v0.5.0 --strategy=resource-merge
 `
 
+var CatShort = `Print the resources in a package`
+var CatLong = `
+  kpt cfg cat [DIR]
```
Comment (frankfarzan) [inline internal/docs/generated/pkgdocs/docs.go:50]: still uses cfg

File: site/reference/pkg/cat/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+Cat is useful for printing only the resources in a package which might
+contain other non-resource files.
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```
+kpt cfg cat [DIR]
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:25]: so you can't cat a specific file?

File: site/reference/pkg/cat/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+# print Resource config from current directory
+kpt cfg cat
+```
+
+<!--mdtogo-->
+
+#### Flags
+
+```sh
+--annotate
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:47]: what does this do? annotate path/index/package annotations? Can you rationalize the use cases for all the flags here?

File: site/reference/pkg/cat/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  annotate resources with their file origins.
+
+--dest string
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:50]: is this required?

File: site/reference/pkg/cat/README.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```
+kpt cfg cat [DIR]
+
+DIR:
+  Path to a package directory. Defaults to the current working directory.
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:28]: Where's the output of this command described?

File: internal/docs/generated/pkgdocs/docs.go (hunk 1 above)
Comment (phanimarupaka) [reply internal/docs/generated/pkgdocs/docs.go]: Updated.

File: site/reference/pkg/cat/README.md (hunk 3 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md:74]: `annotate path/index/package annotations?` => Yes.

Most of the flags try to replicate what `kpt fn source/sink` does. Removed such flags.

File: site/reference/pkg/cat/README.md (hunk 2 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md]: You can. Updated usage.

File: site/reference/pkg/cat/README.md (hunk 4 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md]: No. Functionality is similar to `sink` command. Removed it.

File: site/reference/pkg/cat/README.md (hunk 5 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md:28]: Updated.

File: site/reference/pkg/cat/README.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+### Synopsis
+
+<!--mdtogo:Long-->
+
+```
+kpt pkg cat [FILE | DIR]
+
+DIR:
+  Path to a directory with KRM resource files. Defaults to the current working directory.
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:26]: Descriptions don't match the fact that the argument can be a FILE or DIR. Please do an audit.

File: site/reference/pkg/cat/README.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+FILE:
+  Path to a KRM file.
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:29]: I don't think we use "KRM file" in other places. KRM is obvious to us but necessarily to the end user. As you're auditing the CLI usability and reference docs, make sure we use consistent nomenclature. Related to "config vs resource" discussions.

File: site/reference/pkg/cat/README.md (hunk 3 above)
Comment (frankfarzan) [reply site/reference/pkg/cat/README.md:72]: If source/sink is used instead, what's the use case for keeping annotate flag?

Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md:72]: This will be useful for users to see the file origins.

File: site/reference/pkg/cat/README.md (hunk 7 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md]: renamed it to `resources`. Will make it consistent all across the docs.

File: site/reference/pkg/cat/README.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+type: docs
+description: >
+  Print the resources in a file/directory
+---
+
+<!--mdtogo:Short
+    Print the resources in a file/directory
+-->
+
+Cat prints the resources in a file/directory as yaml to stdout.
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:13]: Style issue: the command name is `cat`, not `Cat`. Proper nouns should be in verbatim quotes.

But in this case, is kind of necessary to name the command:

`Prints resources in a file or directory...`

It's good to start a style guide to make sure we're consistent (Similar to CLI error/validation style guide).

File: site/reference/pkg/cat/README.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+kind: Service
+metadata:
+  name: nginx-service
+```
+
+<!--mdtogo-->
+
+#### Flags
+
+```sh
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:71]: Isn't docs for flags auto-generated from cobra configuration?

File: site/reference/pkg/cat/README.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+--annotate
+  annotate resources with their file origins.
+
+--format
+  format resource before printing. (default true)
+
+--include-local
+  if true, include local-config in the output.
```
Comment (frankfarzan) [inline site/reference/pkg/cat/README.md:79]: What's `local-config`? Do you mean resources with such an annotation? Usage docs have to be precise and have no ambiguity.

Is there a real use case for these two flags?

File: site/reference/pkg/cat/README.md (hunk 8 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md]: Updated.

File: site/reference/pkg/cat/README.md (hunk 9 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md:71]: Yes. This should be cleaned up in cobra as well in kustomize repo in a different PR. https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/internal/commands/cat.go#L35

File: site/reference/pkg/cat/README.md (hunk 10 above)
Comment (phanimarupaka) [reply site/reference/pkg/cat/README.md]: Do you mean resources with such an annotation? => Yes.

I am also not sure if we need them for now. Removed them and we can add later if we need them.

Comment (frankfarzan) [reply site/reference/pkg/cat/README.md]: Yes, when in doubt, remove flags. They have negative costs by causing confusion and polluting the terminal.

**Review by frankfarzan — APPROVED** [review]
Much cleaner

**Pull request merged**